src/libs/SearchUIUtils.ts

```diff
@@ -269,6 +270,79 @@ function isReportActionListItemType(item: SearchListItem): item is ReportActionL
     return reportActionListItem.reportActionID !== undefined;
 }
 
+function isAmountTooLong(amount: number, maxLength = 8): boolean {
+    return Math.abs(amount).toString().length >= maxLength;
+}
+
+function isTransactionAmountTooLong(transactionItem: TransactionListItemType | SearchTransaction | OnyxTypes.Transaction) {
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const amount = Math.abs(transactionItem.modifiedAmount || transactionItem.amount);
+    return isAmountTooLong(amount);
+}
+
+function isTransactionTaxAmountTooLong(transactionItem: TransactionListItemType | SearchTransaction | OnyxTypes.Transaction) {
+    const reportType = (transactionItem as TransactionListItemType)?.reportType;
+    const isFromExpenseReport = reportType === CONST.REPORT.TYPE.EXPENSE;
+    const taxAmount = getTaxAmount(transactionItem, isFromExpenseReport);
+    return isAmountTooLong(taxAmount);
+}
+
+function getWideAmountIndicators(data: TransactionListItemType[] | ReportListItemType[] | TaskListItemType[] | OnyxTypes.SearchResults['data']): {
+    shouldShowAmountInWideColumn: boolean;
+    shouldShowTaxAmountInWideColumn: boolean;
+} {
+    let isAmountWide = false;
+    let isTaxAmountWide = false;
+
+    const processTransaction = (transaction: TransactionListItemType | SearchTransaction) => {
+        isAmountWide ||= isTransactionAmountTooLong(transaction);
+        isTaxAmountWide ||= isTransactionTaxAmountTooLong(transaction);
+    };
+
+    if (Array.isArray(data)) {
+        data.some((item) => {
+            if (isReportListItemType(item)) {
+                const transactions = item.transactions ?? [];
+                for (const transaction of transactions) {
+                    if (isAmountWide && isTaxAmountWide) {
+                        break;
+                    }
+                    processTransaction(transaction);
+                }
+            } else if (isTransactionListItemType(item)) {
+                processTransaction(item);
+            }
+            return isAmountWide && isTaxAmountWide;
+        });
+    } else {
+        Object.entries(data).some(([key, value]) => {
+            if (isTransactionEntry(key)) {
+                processTransaction(value as TransactionListItemType);
+            } else if (isReportEntry(key)) {
+                const report = getReportFromKey(data, key);
+
+                if (!report) {
+                    return isAmountWide && isTaxAmountWide;
+                }
+
+                const transactions = getTransactionsForReport(data, report.reportID);
+                for (const transaction of transactions) {
+                    if (isAmountWide && isTaxAmountWide) {
+                        break;
+                    }
+                    processTransaction(transaction);
+                }
+            }
```

Comment:
Checking again, this whole block seems redundant. The transactions we get from `getTransactionsForReport` are the same ones we are processing in the above if conditions as we are operating on the same `data`